python/cuml/tests/dask/test_dask_logistic_regression.py

```diff
@@ -336,3 +344,60 @@ def test_noreg(fit_intercept, client):
     l1_strength, l2_strength = lr._get_qn_params()
     assert l1_strength == 0.0
     assert l2_strength == 0.0
+
+
+def test_n_classes_small(client):
+    def assert_small(X, y, n_classes):
+        X_df, y_df = _prep_training_data(client, X, y, partitions_per_worker=1)
+        from cuml.dask.linear_model import LogisticRegression as cumlLBFGS_dask
+
+        lr = cumlLBFGS_dask()
+        lr.fit(X_df, y_df)
+        assert lr._num_classes == n_classes
+        return lr
+
+    X = np.array([(1, 2), (1, 3)], np.float32)
+    y = np.array([1.0, 0.0], np.float32)
+    lr = assert_small(X=X, y=y, n_classes=2)
+    assert np.array_equal(
+        lr.classes_.to_numpy(), np.array([0.0, 1.0], np.float32)
+    )
+
+    X = np.array([(1, 2), (1, 3), (1, 2.5)], np.float32)
+    y = np.array([1.0, 0.0, 1.0], np.float32)
+    lr = assert_small(X=X, y=y, n_classes=2)
+    assert np.array_equal(
+        lr.classes_.to_numpy(), np.array([0.0, 1.0], np.float32)
+    )
+
+    X = np.array([(1, 2), (1, 2.5), (1, 3)], np.float32)
+    y = np.array([1.0, 1.0, 0.0], np.float32)
+    lr = assert_small(X=X, y=y, n_classes=2)
+    assert np.array_equal(
+        lr.classes_.to_numpy(), np.array([0.0, 1.0], np.float32)
+    )
+
+    X = np.array([(1, 2), (1, 3), (1, 2.5)], np.float32)
+    y = np.array([10.0, 50.0, 20.0], np.float32)
+    lr = assert_small(X=X, y=y, n_classes=3)
+    assert np.array_equal(
+        lr.classes_.to_numpy(), np.array([10.0, 20.0, 50.0], np.float32)
+    )
+
+
+@pytest.mark.parametrize("fit_intercept", [False, True])
+@pytest.mark.parametrize("n_classes", [2, 8])
```

Comment:
Revised the tests to add n_parts to the argument list. The tests took 18s in total.

Was not able to find a way to combine `parametrize` decorator and `hypothesis strategies` decorator. Should be fine assuming the 18s is acceptable. 

the `penalty` argument will be added to tests in PR [5587](https://github.com/rapidsai/cuml/pull/5587), a followup that supports "l1" and "elasticnet" penalty.

Reply:
It's not possible to use `pytest.mark.parametrize` and `given()` in combination. Typically one would provide more general strategies and then use hypothesis examples to test specific test cases. Here is an example for this:
https://github.com/rapidsai/cuml/blob/3e5c8e9d147257563ca987c3e7a10145ba1cda4f/python/cuml/tests/test_linear_model.py#L333-L342

It would be good to test both supported and non-supported penalty arguments here as well to check that the interface behaves as expected, i.e., provides user-friendly error messages and doesn't just break. This also makes it easier for us to detect breaking changes at a later stage.

Reply:
Added the tests for "l1" and "elasticnet". When they are used, the class throws an error with message.